client/src/containers/Dataset.jsx

```diff
@@ -4,33 +4,57 @@ import Immutable from 'immutable';
 import { connect, useSelector } from 'react-redux';
 import { withRouter } from 'react-router';
 import { showModal } from '../actions/activeModal';
-import { fetchDataset, updateDatasetMeta, pollTxImportStatus, startTx, undoTx, endTx } from '../actions/dataset';
+import {
+  fetchDataset,
+  fetchDatasetGroups,
+  updateDatasetMeta,
+  pollTxImportStatus,
+  startTx,
+  undoTx,
+  endTx,
+} from '../actions/dataset';
 import { showNotification } from '../actions/notification';
 import { getId, getTitle } from '../domain/entity';
-import { getTransformations, getRows, getColumns, getIsLockedFromTransformations } from '../domain/dataset';
+import {
+  getTransformations,
+  getRows,
+  getColumns,
+  getIsLockedFromTransformations,
+} from '../domain/dataset';
 import * as api from '../utilities/api';
 import LoadingSpinner from '../components/common/LoadingSpinner';
 import { TRANSFORM_DATASET } from '../constants/analytics';
 import { trackEvent, trackPageView } from '../utilities/analytics';
 import NavigationPrompt from '../components/common/NavigationPrompt';
 import DatasetHeader from '../components/dataset/DatasetHeader';
-import DatasetTable from '../components/dataset/DatasetTable';
+import DatasetTableV2 from '../components/dataset/DatasetTableV2';
+import DatasetTableV1 from '../components/dataset/DatasetTable';
 import usePendingSaving from '../components/common/PendingSaving';
 
 require('../components/dataset/Dataset.scss');
 
 function Dataset(props) {
-  const dataset = useSelector(state => state.library.datasets[props.params.datasetId]);
+  const [isFeatureFlag] = useSelector(
+    state => state.env.environment.rqg
+  );
```

Comment:
@tangrammer is this the correct way of getting feature flag. Is there a possibility it'll be undefined?